numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1074,33 +972,99 @@ array_richcompare(PyArrayObject *self, PyObject *other, int cmp_op)
         Py_INCREF(Py_NotImplemented);
         return Py_NotImplemented;
     }
-    if (result == NULL) {
+
+    /*
+     * At this point we are locked into comparing (both are arrays) or
+     * something we would have tried to convert.
```

Comment:
I don't understand the comment above: is the closing parenthesis in the wrong place? If we tried to convert it, did we fail? I would prefer it said this, but maybe it is not true.
```suggestion
     * At this point we comparing two ndarrays.
```